project/src/backend/sdl/SDLApplication.cpp

```diff
@@ -923,6 +932,83 @@ namespace lime {
 	}
 
 
+	int SDLApplication::BatchUpdate (int numEvents) {
+
+		SDL_Event* mouseMoved = NULL;
+
+		if (!active) {
+			return -1;
+		}
+		
+		queueLength = numEvents;
+
+		if (queueLength > queueMaxLength) {
+
+			if (NULL != eventQueue) {
+				SDL_free (eventQueue);
+			}
+
+			eventQueue = reinterpret_cast<SDL_Event*> (SDL_malloc (queueLength * sizeof (SDL_Event)));
+			if (NULL == eventQueue) {
+				queueLength = 0;
+				queueMaxLength = 0;
+			} else {
+				queueMaxLength = queueLength;
```

Comment:
I feel like `queueMaxLength` is the wrong name if you can just increase it. From the name, I'd expect excess events to be discarded or something.

Actually, what is the `queueLength` variable even for? It stores the last value of `numEvents`, but it's never needed outside this function. We could have `queueLength` do what `queueMaxLength` is currently doing, and use `numEvents` as the number of events. (That is, find and replace "queueLength" → "numEvents", then replace "queueMaxLength" → "queueLength".)